### DISCUSSION TITLE
Header at the top of the lock file

### DISCUSSION THREAD

**Discussion by akhileshraju** [python-poetry/poetry#3954] [Q&A] [answered]
At the place I work, we are looking to open source a py package.
Our legal team requires that we add license headers to all the files in our repo. We use poetry and I am not sure if it’s safe to manually add a bunch of lines to the beginning of the lock file.
Does poetry ignore lines in the lock file that start with some particular character like #?

**Comment by abn** [comment, marked as answer]
Since the lock file is a toml file, syntactically comments are fine. However, since the lock file is expected to be completely poetry managed and because we rewrite the lock file instead of preserving any comments or manual edits in most cases, there is no guarantee that these comments will be preserved for all operations. For example if a new dependency is added or old one removed, you will end up with a brand new toml file. You will have to re-add the licence everytime, or alternatively use a pre-commit hook to ensure a licence header is injected into all files automatically.

**Reply by akhileshraju** [reply]
That makes sense. Thanks for the quick response @abn 😀
